typhon/plugins/core.py

```diff
@@ -0,0 +1,198 @@
+############
+# Standard #
+############
+import sys
+import logging
+import importlib
+
+###############
+# Third Party #
+###############
+import numpy as np
+from pydm.data_plugins.plugin import PyDMConnection
+from pydm.PyQt.QtCore import pyqtSlot, Qt
+
+##########
+# Module #
+##########
+
+logger = logging.getLogger(__name__)
+
+
+class ClassConnection(PyDMConnection):
+    """
+    Connection which spawns an object of specified class
+
+    Attributes
+    ----------
+    obj: object
+        The created object is stored on the connection for use by subclasses
+
+    Example
+    -------
+    .. code:: python
+
+        conn = ClassConnection("cls://ophyd.EpicsMotor|'Tst:Mtr:07'|name=Test",
+                               "ophyd.EpicsMotor|'Tst:Mtr:07'|name=Test")
+
+    Notes
+    -----
+    "preassembled" provides a shortcut instead of creating a new object. This
+    is useful if channel connections are being scripted programatically as
+    opposed to from the Designer. The channel and address arguments are
+    completely ignored in this case.
+    """
+    def __init__(self, channel, address, protocol=None,
+                 parent=None, preassembled=None):
+        # Base initialization
+        super().__init__(channel, address, protocol=protocol, parent=parent)
+        # Just use the preassambled object if given to us
+        # Otherwise, instantiate our own
+        logger.debug("Creating connection to %s", address)
+        if preassembled:
+            logger.debug("Using an already instantiated object as reference")
+            self.obj = preassembled
+        else:
+            # NOTE: This was taken entirely from happi/loader.py. The reason it
```

Comment:
Could this then be brought out to a supporting function? The parsing probably should not be buried in this `__init__` (even if it is copied from `happi`) and tested separately.

If these functions are ever moved up into a shared package, maybe then a grammar to parse would be a good idea (current one is problematic with no way to escape`|`, commas/equals in strings, etc...)

Reply:
@klauer Thanks for calling me out on this. I think I'll just add a dependency on `happi` where this function is tested thoroughly. If someone really wants to use `typhon` but not incur the innocuous `happi` dependency we'll cross that bridge then.

Reply:
I looked at this again, and there are actually some deep seeded differences that I forgot about. I refactored the object instantiation out of the class constructor and into a new function and added a test.